### PR TITLE
changed all instances of exit(1) to click.abort()

### PR DESCRIPTION
**Pull request by AlexVCS**
working on #67

### CODE REVIEW

**Comment by calebevans** [comment]
@AlexVCS Thanks for the PR, just a heads-up, you are probably going to need to import click to many of these files. It also looks like mypy does not accept the click.Abort() as an exit or "return" statement.

**Comment by AlexVCS** [comment]
> @AlexVCS Thanks for the PR, just a heads-up, you are probably going to need to import click to many of these files. It also looks like mypy does not accept the click.Abort() as an exit or "return" statement.

what do you mean by need to import click to many of the files? I'm not familiar with Python just was trying to help.

**Comment by AlexVCS** [comment]
thank you, I think the files that need it should have it now

File: cli/objects/configuration.py [hunk 1]
```diff
@@ -77,7 +78,7 @@ def _get_rules(self, rules_json: list[dict[Any, Any]]) -> Optional[list[Rule]]:
             self.logger.error(
                 "Firewatch config is empty, please populate the configuration and try again.",
             )
-            exit(1)
+            click.abort()
```
Comment (myakove) [inline cli/objects/configuration.py:81]: click.Abort() is an exception, you need to raise it.

**Review by myakove — CHANGES_REQUESTED** [review]

File: cli/objects/configuration.py [hunk 2]
```diff
@@ -95,7 +96,7 @@ def _get_default_jira_project(self) -> str:
             self.logger.error(
                 "Environment variable $FIREWATCH_DEFAULT_JIRA_PROJECT is not set, please set the variable and try again.",
             )
-            exit(1)
+            click.abort()
```
Comment (myakove) [inline cli/objects/configuration.py:99]: Needs to rise in all places.

**Review by myakove — CHANGES_REQUESTED** [review]

**Comment by myakove** [comment]
And you need to install pre-commit in order to run the pre-commit tests before pushing a PR

**Comment by AlexVCS** [comment]
> And you need to install pre-commit in order to run the pre-commit tests before pushing a PR

I installed the dependencies, but when I run `make pre-commit` I get this error:

`pre-commit run --all-files
make: pre-commit: No such file or directory
make: *** [pre-commit] Error 1`

I'm googling but any ideas of what I need to do?

File: cli/objects/configuration.py [hunk 3]
```diff
@@ -77,7 +78,7 @@ def _get_rules(self, rules_json: list[dict[Any, Any]]) -> Optional[list[Rule]]:
             self.logger.error(
                 "Firewatch config is empty, please populate the configuration and try again.",
             )
-            exit(1)
+            raise click.abort()
```
Comment (myakove) [inline cli/objects/configuration.py:81]:
```suggestion
            raise click.Abort()
```

**Review by myakove — CHANGES_REQUESTED** [review]

**Comment by myakove** [comment]
While you inside the repository directory on your local PC run `pre-commit install`
After that pre-commit should be run automatically when you do `git commit......`

**Pull request closed without merging**